### PR TITLE
Add new TPUv2 product and VM resource in beta

### PR DESCRIPTION
**Pull request by zeleena**
Add new TPUv2 product and required VM (TPU VM) resource with minimum required fields and description to the google-beta provider. Also add data sources for the required fields in beta.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_tpu_v2_vm` (beta)
```

### CODE REVIEW

File: mmv1/products/tpuv2/Vm.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+name: 'Vm'
+description: |
+  A Cloud TPU VM instance.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/tpu/docs/'
+  api: 'https://cloud.google.com/tpu/docs/reference/rest/v2/projects.locations.nodes'
+base_url: projects/{{project}}/locations/{{zone}}/nodes
+self_link: projects/{{project}}/locations/{{zone}}/nodes/{{name}}
+create_url: projects/{{project}}/locations/{{zone}}/nodes?nodeId={{name}}
```
Comment (anejavipul) [inline mmv1/products/tpuv2/Vm.yaml:24]: Where is the release track configured for this endpoint, ie we want to hit v2alpha1 proto

File: mmv1/products/tpuv2/product.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: TpuV2
+display_name: Cloud TPU v2
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://tpu.googleapis.com/v2/
```
Comment (anejavipul) [inline mmv1/products/tpuv2/product.yaml:20]: should this be v2apha1 ?

Comment (zeleena) [reply mmv1/products/tpuv2/product.yaml]: Done

File: mmv1/products/tpuv2/Vm.yaml (hunk 1 above)
Comment (zeleena) [reply mmv1/products/tpuv2/Vm.yaml:24]: Done, I updated it in `mmv1/products/tpuv2/product.yaml` per your other comment.

File: mmv1/products/tpuv2/product.yaml (hunk 2 above)
Comment (zeleena) [reply mmv1/products/tpuv2/product.yaml:20]: In an offline conversation with @rileykarson, the Terraform team would prefer to have the v2 version in the google-beta provider and then we can override with v2alpha1 when we implement the feature.

**Review by trodge — CHANGES_REQUESTED** [review]
Could you also add an [update test](https://googlecloudplatform.github.io/magic-modules/develop/test/#add-an-update-test)?

**Comment by zeleena** [comment]
@trodge We only update the description field for now as we'd like to start by getting a minimal feature set checked in, but I can look into adding it.

**Review by trodge — APPROVED** [review]

**Pull request merged**